tests/response/test_base_response.py

```diff
@@ -0,0 +1,178 @@
+from typing import Any, Optional
+
+import pytest
+from pydantic_openapi_schema.v3_1_0 import Info, OpenAPI
+
+from starlite import MediaType, OpenAPIMediaType, get
+from starlite.response import Response
+from starlite.status_codes import (
+    HTTP_100_CONTINUE,
+    HTTP_101_SWITCHING_PROTOCOLS,
+    HTTP_102_PROCESSING,
+    HTTP_103_EARLY_HINTS,
+    HTTP_200_OK,
+    HTTP_204_NO_CONTENT,
+    HTTP_500_INTERNAL_SERVER_ERROR,
+)
+from starlite.testing import create_test_client
+from starlite.types import Empty
+
+
+def test_response_headers() -> None:
+    @get("/")
+    def handler() -> Response:
+        return Response(content="hello world", media_type=MediaType.TEXT, headers={"first": "123", "second": 456})
+
+    with create_test_client(handler) as client:
+        response = client.get("/")
+        assert response.headers["first"] == "123"
+        assert response.headers["second"] == "456"
+        assert response.headers["content-length"] == "11"
+        assert response.headers["content-type"] == "text/plain; charset=utf-8"
+
+
+def test_set_cookie() -> None:
+    @get("/")
+    def handler() -> Response:
+        response = Response(content=None)
+        response.set_cookie(
+            "test",
+            "abc",
+            max_age=60,
+            expires=60,
+            path="/",
+            secure=True,
+            httponly=True,
+            samesite="lax",
+        )
+        assert len(response.cookies) == 1
+        return response
+
+    with create_test_client(handler) as client:
+        response = client.get("/")
+        assert response.cookies.get("test") == "abc"
+
+
+def test_delete_cookie() -> None:
+    @get("/create")
+    def create_cookie_handler() -> Response:
+        response = Response(content=None)
+        response.set_cookie(
+            "test",
+            "abc",
+            max_age=60,
+            expires=60,
+            path="/",
+            secure=True,
+            httponly=True,
+            samesite="lax",
+        )
+        assert len(response.cookies) == 1
+        return response
+
+    @get("/delete")
+    def delete_cookie_handler() -> Response:
+        response = Response(content=None)
+        response.delete_cookie(
+            "test",
+            "abc",
+        )
+        assert len(response.cookies) == 1
+        return response
+
```

Comment:
Since values don't matter for cookie deletion, maybe add a test for that case. I.e. that a cookie is deleted if its key matches but its value does not.

Reply:
can do. The point really is what makes the browser delete the cookie, which is not the value but rather the key+path+domain combination.

Reply:
Correct. I just think we should have a test that checks that we are doing it on our side as well. What I mean is:

Test that `response.delete_cookie("test, "abc")` should deletes any cookie that matches these parameters, but ignore things like its value.